File: Java/src/main/java/Solution301_400/Solution387.java
package Solution301_400;

import java.util.HashMap;
import java.util.HashSet;

/**
 * @author windows
 */
public class Solution387 {
    public int firstUniqChar(String s) {
        int[] chars = new int[26];
        for (int i = 0; i < s.length(); i++) {
            chars[s.charAt(i) - 'a']++;
        }
        for (int i = 0; i < s.length(); i++) {
            if (chars[s.charAt(i) - 'a'] == 1){
                return i;
            }
        }
        return -1;
    }


    public static void main(String[] args) {
        Solution387 solution387 = new Solution387();
    }
}

File: C++/src/Solution301_400/Solution387.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution387.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <unordered_map>

int Solution387::firstUniqChar(string s) {
    unordered_map<char, int> map;
    for (char c : s) {
        map[c]++;
    }
    for (int i = 0; i < s.size(); ++i) {
        if (map[s[i]] == 1){
            return i;
        }
    }
    return -1;
}
